File: common/src/main/java/com/pluxity/asset/service/AssetValidator.java
package com.pluxity.asset.service;

import static com.pluxity.global.constant.ErrorCode.NOT_FOUND_ASSET;

import com.pluxity.asset.repository.AssetRepository;
import com.pluxity.global.exception.CustomException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AssetValidator {
    private final AssetRepository assetRepository;

    @Transactional(readOnly = true)
    public void validateAssetId(Long id) {
        assetRepository.findById(id).orElseThrow(() -> new CustomException(NOT_FOUND_ASSET, id));
    }
}

File: common/src/main/java/com/pluxity/asset/service/AssetValidator.kt
package com.pluxity.asset.service

import com.pluxity.asset.repository.AssetRepository
import com.pluxity.global.constant.ErrorCode
import com.pluxity.global.exception.CustomException
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AssetValidator(
    private val assetRepository: AssetRepository,
) {
    @Transactional(readOnly = true)
    fun validateAssetId(id: Long) {
        assetRepository.findByIdOrNull(id) ?: throw CustomException(ErrorCode.NOT_FOUND_ASSET, id)
    }
}
